### PR TITLE
samples: boards: nordic: system_off: Add GRTC wakeup

### PR DESCRIPTION
**Pull request by nordic-babu**
Extend system_off sample to use GRTC timer as a wake up source.

### CODE REVIEW

**Comment by nordic-babu** [comment]
FYI @adamkondraciuk @kl-cruz

**Review by nordic-bami — APPROVED** [review]

**Review by nordic-piks — APPROVED** [review]

**Review by adamkondraciuk — APPROVED** [review]

**Review by carlescufi — DISMISSED** [review]

File: samples/boards/nordic/system_off/Kconfig [hunk 1]
```diff
@@ -21,4 +21,10 @@ config APP_USE_RETAINED_MEM
 
 endchoice
 
+config GRTC_WAKEUP_ENABLE
+	bool "Use GRTC to wake up device from system off"
+	default n
```
Comment (anangl) [inline samples/boards/nordic/system_off/Kconfig:26]: `default n` is redundant

File: samples/boards/nordic/system_off/src/main.c [hunk 2]
```diff
@@ -16,7 +16,12 @@
 #include <zephyr/sys/poweroff.h>
 #include <zephyr/sys/util.h>
 
+#if IS_ENABLED(CONFIG_GRTC_WAKEUP_ENABLE)
```
Comment (anangl) [inline samples/boards/nordic/system_off/src/main.c:19]: https://github.com/zephyrproject-rtos/zephyr/blob/6e653a96f7903fc932575b37f449ceaf4e94c3e7/include/zephyr/sys/util_macro.h#L133-L134

File: samples/boards/nordic/system_off/Kconfig (hunk 1 above)
Comment (nordic-babu) [reply samples/boards/nordic/system_off/Kconfig]: Thanks, removed

File: samples/boards/nordic/system_off/src/main.c (hunk 2 above)
Comment (nordic-babu) [reply samples/boards/nordic/system_off/src/main.c]: Switched to pure `#if defined()`

**Review by carlescufi — APPROVED** [review]

**Review by anangl — APPROVED** [review]

**Pull request merged**